File: src/order/application/ViewOrder.ts
import { Order } from "../domain/Order.js";
import { OrderRepository } from "../domain/OrderRepository.js";

/**
 * ViewOrder — Application Service (Use Case)
 *
 * Retrieves a single order by ID.
 */

export class ViewOrder {
  constructor(private readonly orderRepository: OrderRepository) {}

  async execute(orderId: string): Promise<Order> {
    const order = await this.orderRepository.findById(orderId);
    if (!order) {
      throw new Error(`Order not found: ${orderId}`);
    }
    return order;
  }
}

File: src/order/application/view_order.py
"""ViewOrder use case — retrieve a single order by ID."""

from __future__ import annotations

from src.order.domain.order_repository import OrderRepository


class ViewOrder:
    """Application service that handles the 'view order' use case."""

    def __init__(self, repository: OrderRepository) -> None:
        self._repository = repository

    def execute(self, order_id: str) -> dict:
        """Look up an order and return a JSON-serialisable response dict."""
        order = self._repository.find_by_id(order_id)
        if order is None:
            return {
                "status": "error",
                "error": f"Order not found: {order_id}",
            }
        return {"status": "success", "order": order.to_dict()}
